packages/vsx-registry/src/browser/vsx-extension.tsx

```diff
@@ -100,6 +104,56 @@ export class VSXExtension implements VSXExtensionData, TreeElement {
 
     protected readonly data: Partial<VSXExtensionData> = {};
 
+    protected _body: HTMLDivElement | undefined;
```

Comment:
It should be replaced with:
```
protected _editor: VSXExtensionEditorComponent | undefined;

renderEditor(): React.ReactNode {
        return <VSXExtensionEditorComponent extension={this} ref={ref = (this._editor = ref || undefined)} />;
    }
```

Reply:
The `_body` member was kept from master (but just moved up a couple lines to be more visible) since the `openLink()` method on line 531 uses this in some logic.

Reply:
Hi @akosyakov, the reason I have these additional (`body`, `header`, and `scrollContainer`) elements is because it seems that the absence of a `readme` on line 475 could potentially cause the `body` and `scroll-container` to not be rendered. Additionally because there is manipulation of the virtual DOM and I need a ref to all 3 elements in `setContainerHeights` and `updateOnResize` to keep track of each individually.

Reply:
My problem is not that you add them, but how, i.e. there are public APIs to allow setting them by child components. Instead proper encapsulation should be applied, i.e. they should be exposed on child components instead and here we should only capture a child component ref::
```ts
const body = this._editor?.body;
```